include/cantera/thermo/IdealSolidSolnPhaseTabulatedThermo.h

```diff
@@ -0,0 +1,135 @@
+/*
+ * IdealSolidSolnPhaseTabulatedThermo.h
+ *
+ *  Created on: 04.06.2016
+ *      Author: mmayur
+ */
+
+/**
+ *  @file IdealSolidSolnPhaseTabulatedThermo.h
+ * Definition file for a derived class of ThermoPhase that assumes either
+ * an ideal gas or ideal solution approximation and handles
+ * variable pressure standard state methods for calculating
+ * thermodynamic properties (see \ref thermoprops and
+ * class \link Cantera::IdealSolidSolnPhaseTabulatedThermo IdealSolidSolnPhaseTabulatedThermo\endlink).
+ */
+/*
+ * Copyright (2005) Sandia Corporation. Under the terms of
+ * Contract DE-AC04-94AL85000 with Sandia Corporation, the
+ * U.S. Government retains certain rights in this software.
+ */
+#ifndef CT_IDEALSOLIDSOLNPHASETABULATEDTHERMO_H
+#define CT_IDEALSOLIDSOLNPHASETABULATEDTHERMO_H
+
+#include "IdealSolidSolnPhase.h"
+#include "cantera/base/utilities.h"
+#include <vector>
+#include <iterator>
+#include <iostream>
+#include <fstream>
+
+namespace Cantera
+{
+
+//!   Overloads the virtual methods of class ThermoPhase to implement the
+//!   incompressible equation of state.
+/**
+ * <b> Specification of Solution Thermodynamic Properties </b>
+ *
+ * The density is assumed to be constant, no matter what the concentration of the solution.
+ *
+ * @ingroup thermoprops
+ */
+class IdealSolidSolnPhaseTabulatedThermo : public IdealSolidSolnPhase
+{
+public:
+    //! Constructor.
+	IdealSolidSolnPhaseTabulatedThermo() {}
+
+    //! Copy Constructor
+    /*!
+     * @param right Object to be copied
+     */
+	IdealSolidSolnPhaseTabulatedThermo(const IdealSolidSolnPhaseTabulatedThermo& right);
+
+    //! Assignment Operator
+    /*!
+     * @param right Object to be copied
+     */
+	IdealSolidSolnPhaseTabulatedThermo& operator=(const IdealSolidSolnPhaseTabulatedThermo& right);
+
+    //! Duplication routine for objects which inherit from ThermoPhase
+    /*!
+     *  This virtual routine can be used to duplicate objects
+     *  derived from ThermoPhase even if the application only has
+     *  a pointer to ThermoPhase to work with.
+     */
+    virtual ThermoPhase* duplMyselfAsThermoPhase() const;
+
+    virtual std::string type() const {
+        return "IdealSolidSolutionTabultedThermo";
+    }
+
+    //! Was added because of the DENIS LFP Runaway simulation.

```

Comment:
Can you please explain a little more what this comment means? Remember that the comments should make sense to a developer who hasn't looked at the code before, but who needs to maintain it. What is the DENIS LFP Runaway simulation? Why is it important? How does this function help?